src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -150,6 +158,57 @@ class WorkspaceService(dataSource: DataSource, workspaceDAO: WorkspaceDAO, entit
       }
     }
 
+  def copyEntities(entityCopyDef: EntityCopyDefinition, uri: Uri): PerRequestMessage =
+    dataSource inTransaction { txn =>
+      val destNamespace = entityCopyDef.destinationWorkspace.namespace
+      val destWorkspace = entityCopyDef.destinationWorkspace.name
+
+      val sourceNamespace = entityCopyDef.sourceWorkspace.namespace
+      val sourceWorkspace = entityCopyDef.sourceWorkspace.name
+
+      val entityNames = entityCopyDef.entityNames
+      val entityType = entityCopyDef.entityType
+
+      val conflicts = entityDAO.copyEntities(destNamespace, destWorkspace, sourceNamespace, sourceWorkspace, entityType, entityNames, txn)
+
+      conflicts.size match {
+        case 0 => {
+          RequestComplete(StatusCodes.Created, workspaceDAO.load(destNamespace, destWorkspace, txn))
```

Comment:
there wasnt a great reason to return workspace, could be changed to the entities from the dest workspace i guess?